server/errorhandler.go

```diff
@@ -0,0 +1,47 @@
+package server
+
+import (
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/containous/traefik/middlewares"
+)
+
+// RecordingErrorHandler is an error handler, implementing the vulcand/oxy
+// error handler interface, which is recording network errors by setting the
+// request context value with the key middlewares.DefaultNetErrCtxKey to true.
+// This enables the retry middleware to only retry on actual network errors.
+type RecordingErrorHandler struct{}
+
+// NewRecordingErrorHandler creates and returns a new instance of RecordingErrorHandler.
+func NewRecordingErrorHandler() *RecordingErrorHandler {
```

Comment:
After my refactoring it has a property now and I would leave the `New..` function.